Variational_medial_axis/include/CGAL/variational_medial_axis_sampling.h

```diff
@@ -0,0 +1,1610 @@
+          typename VertexPointMap_ = Default>
+class Variational_medial_axis
+{
+private:
+  using VPM = typename Default::Get<VertexPointMap_,
+                                    typename boost::property_map<TriangleMesh_, vertex_point_t>::const_type>::type;
+  using GT = typename Default::Get<
+      GeomTraits_,
+      typename Kernel_traits<typename boost::property_traits<
+          typename boost::property_map<TriangleMesh_, vertex_point_t>::type>::value_type>::Kernel>::type;
+
+  using FT = typename GT::FT;
+  using Point_3 = typename GT::Point_3;
+  using Vector_3 = typename GT::Vector_3;
+  using Sphere_3 = typename GT::Sphere_3;
+  using MSMesh = Medial_Sphere_Mesh<TriangleMesh_, GT>;
+  using MSphere = typename MSMesh::MSphere;
+  using Sphere_ID = typename MSMesh::Sphere_ID;
+
+  using Tree = AABB_tree<AABB_traits_3<GT, AABB_face_graph_triangle_primitive<TriangleMesh_, VPM>>>;
+
+  using face_descriptor = typename boost::graph_traits<TriangleMesh_>::face_descriptor;
+  using edge_descriptor = typename boost::graph_traits<TriangleMesh_>::edge_descriptor;
+  using vertex_descriptor = typename boost::graph_traits<TriangleMesh_>::vertex_descriptor;
+
+  // Property map types
+  using Vertex_normal_tag = CGAL::dynamic_vertex_property_t<Vector_3>;
+  using Vertex_normal_map = typename boost::property_map<TriangleMesh_, Vertex_normal_tag>::const_type;
+  using Vertex_area_tag = CGAL::dynamic_vertex_property_t<FT>;
+  using Vertex_area_map = typename boost::property_map<TriangleMesh_, Vertex_area_tag>::const_type;
+  using Face_normal_tag = CGAL::dynamic_face_property_t<Vector_3>;
+  using Face_normal_map = typename boost::property_map<TriangleMesh_, Face_normal_tag>::const_type;
+  using Face_area_tag = CGAL::dynamic_face_property_t<FT>;
+  using Face_area_map = typename boost::property_map<TriangleMesh_, Face_area_tag>::const_type;
+  using Vertex_error_tag = CGAL::dynamic_vertex_property_t<FT>;
+  using Vertex_error_map = typename boost::property_map<TriangleMesh_, Vertex_error_tag>::const_type;
+  using Vertex_cluster_sphere_tag = CGAL::dynamic_vertex_property_t<Sphere_ID>;
+  using Vertex_cluster_sphere_map = typename boost::property_map<TriangleMesh_, Vertex_cluster_sphere_tag>::const_type;
+  using Vertex_medial_sphere_pos_tag = CGAL::dynamic_vertex_property_t<Point_3>;
+  using Vertex_medial_sphere_pos_map =
+      typename boost::property_map<TriangleMesh_, Vertex_medial_sphere_pos_tag>::const_type;
+  using Vertex_medial_sphere_radius_tag = CGAL::dynamic_vertex_property_t<FT>;
+  using Vertex_medial_sphere_radius_map =
+      typename boost::property_map<TriangleMesh_, Vertex_medial_sphere_radius_tag>::const_type;
+
+public:
+  /// \name Constructor
+  ///@{
+  ///
+  /// The constructor of a vmas object.
+  ///
+  /// @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  ///
+  /// @param tmesh
+  ///        The input triangle mesh without borders.
+  /// @param vpm
```

Comment:
can be passed as a named parameter (the type must be the same as the template parameter of course)